### PR TITLE
ED-4516: Cantina Audit(14): add finality delay for bedrock

### PR DESCRIPTION
**Pull request by johnwhitton**

### CODE REVIEW

File: contracts/Prover.sol [hunk 1]
```diff
@@ -336,6 +337,15 @@ contract Prover is SimpleProver {
         require(
             existingSettlementBlockProof.stateRoot == l1WorldStateRoot, "settlement chain state root not yet proved"
         );
+
+        // check that the End Batch Block timestamp is greater than the current timestamp + finality delay
+        uint256 endBatchBlockTimeStamp = _bytesToUint(RLPReader.readBytes(RLPReader.readList(rlpEncodedBlockData)[11]));
+
+        require(
```
Comment (StoyanD) [inline contracts/Prover.sol:344]: y not use custom errors instead of requires?

File: contracts/Prover.sol [hunk 2]
```diff
@@ -499,6 +515,9 @@ contract Prover is SimpleProver {
         FaultDisputeGameProofData memory faultDisputeGameProofData,
         bytes32 l1WorldStateRoot
     ) public {
+        require(
```
Comment (StoyanD) [inline contracts/Prover.sol:518]: cleaner if u use a modifier for this

Comment (StoyanD) [reply contracts/Prover.sol:518]: u should add tests for this condition

File: contracts/Prover.sol (hunk 1 above)
Comment (StoyanD) [reply contracts/Prover.sol:344]: u should test this condition

**Review by StoyanD — CHANGES_REQUESTED** [review]
lg, can u change some requires to modifiers and add tests for the new conditions your checking for?

File: contracts/Prover.sol (hunk 1 above)
Comment (johnwhitton) [reply contracts/Prover.sol]: nice catch 🙏 
updated now

File: contracts/Prover.sol (hunk 2 above)
Comment (johnwhitton) [reply contracts/Prover.sol]: nice catch updated now 🙏

**Comment by johnwhitton** [comment]
Yep have removed all requires and added a modifier 🙏

**Review by StoyanD — APPROVED** [review]

**Pull request merged**